### PR TITLE
Fix compilation issues in VS2015

### PR DESCRIPTION
**Pull request by langazov**

### CODE REVIEW

File: wsgate/Primary.cpp [hunk 1]
```diff
@@ -40,28 +40,28 @@ namespace wsgate {
...
-        rdp->update->primary->Mem3Blt = cbMem3Blt;
-        rdp->update->primary->SaveBitmap = cbSaveBitmap;
-        rdp->update->primary->GlyphIndex = cbGlyphIndex;
-        rdp->update->primary->FastIndex = cbFastIndex;
-        rdp->update->primary->FastGlyph = cbFastGlyph;
-        rdp->update->primary->PolygonSC = cbPolygonSC;
-        rdp->update->primary->PolygonCB = cbPolygonCB;
-        rdp->update->primary->EllipseSC = cbEllipseSC;
-        rdp->update->primary->EllipseCB = cbEllipseCB;
+        rdp->update->primary->DstBlt =reinterpret_cast<pDstBlt>(cbDstBlt);
```
Comment (alexpilotti) [inline wsgate/Primary.cpp]: nit: missing space

File: wsgate/RDP.cpp [hunk 2]
```diff
@@ -389,6 +389,7 @@ namespace wsgate {
             m_rdpSettings->PreconnectionBlob = strdup(pcb.c_str());
         }
 
+		m_rdpSettings->SoftwareGdi = true;
```
Comment (alexpilotti) [inline wsgate/RDP.cpp]: please replace tab with spaces

File: wsgate/RDP.cpp [hunk 3]
```diff
@@ -844,11 +845,11 @@ namespace wsgate {
...
         memset(&p, 0, sizeof(p));
         p.size = sizeof(MyPointer);
-        p.New = cbPointer_New;
-        p.Free = cbPointer_Free;
-        p.Set = cbPointer_Set;
-        p.SetNull = cbPointer_SetNull;
-        p.SetDefault = cbPointer_SetDefault;
+        p.New = reinterpret_cast<pPointer_New>(cbPointer_New);
+        p.Free = reinterpret_cast<pPointer_Free>(cbPointer_Free);
+        p.Set =reinterpret_cast<pPointer_Set>(cbPointer_Set);
```
Comment (alexpilotti) [inline wsgate/RDP.cpp]: ditto

File: wsgate/RDP.cpp [hunk 4]
```diff
@@ -844,11 +845,11 @@ namespace wsgate {
...
+        p.SetNull =reinterpret_cast<pPointer_SetNull>(cbPointer_SetNull);
```
Comment (alexpilotti) [inline wsgate/RDP.cpp]: ditto

**Comment by alexpilotti** [comment]
Looking good, if you could please fix those small formatting nits, it's good to merge

**Comment by langazov** [comment]
Hello , 
I haven't contributed to other projects yet,
can you please advise what will be best way for you to do it ?
Do I have to make a new  pull request ?
Sorry if my question sounds dumb, just new to this :)

**Comment by c64cosmin** [comment]
@langazov There are two ways:
You could add a new commit with the changes, and the pull request will be updated automatically (pull requests keep track of a certain branch, in your case is master)
Best way is to use rebasing:
For example if I want to go back in time and change some commits, let's say 2 commits I will execute
`git rebase -i HEAD~2`
(-i stands for interactive, it will show up a editable text file where you can cherry pick the commits and what to do with them
 you will want to edit them, thus 'edit' or 'e' will be added)

now by using
`git log` or 'git show' you can see on which commit you are on
Edit the files and then commit.

After the commit is replaced, you will want to continue rebasing, you will use
`git rebase --continue` to go to the next commit
you can also use if you are under a commit that you want to skip
`git rebase --skip`

After rebasing has been done, you will need to overwrite the old commits, by force pushing to your repository
`git push -f <remote> <branch>`

**Comment by langazov** [comment]
Thank you very much, 
I will edit them little later today :)

**Pull request merged**